runners/direct-java/src/main/java/org/apache/beam/runners/direct/BoundedReadEvaluatorFactory.java

```diff
@@ -38,64 +39,44 @@
+  }
+
+  @Override
+  public Collection<CommittedBundle<?>> getInitialInputs(AppliedPTransform<?, ?, ?> transform) {
+    return createInitialSplits((AppliedPTransform) transform);
+  }
+
+  private <OutputT> Collection<CommittedBundle<?>> createInitialSplits(
+      AppliedPTransform<?, ?, Read.Bounded<OutputT>> transform) {
+    BoundedSource<OutputT> source = transform.getTransform().getSource();
+    return Collections.<CommittedBundle<?>>singleton(
+        evaluationContext
+            .<BoundedSourceShard<OutputT>>createRootBundle()
+            .add(WindowedValue.valueInGlobalWindow(BoundedSourceShard.of(source)))
+            .commit(BoundedWindow.TIMESTAMP_MAX_VALUE));
   }
 
   @SuppressWarnings({"unchecked", "rawtypes"})
   @Override
   @Nullable
   public <InputT> TransformEvaluator<InputT> forApplication(
-      AppliedPTransform<?, ?, ?> application, @Nullable CommittedBundle<?> inputBundle)
-      throws IOException {
-    return getTransformEvaluator((AppliedPTransform) application);
+      AppliedPTransform<?, ?, ?> application, CommittedBundle<?> inputBundle) throws IOException {
+    return createEvaluator((AppliedPTransform) application);
   }
 
-  @Override
-  public void cleanup() {}
-
-  /**
-   * Get a {@link TransformEvaluator} that produces elements for the provided application of {@link
-   * Bounded Read.Bounded}, initializing the queue of evaluators if required.
-   *
-   * <p>This method is thread-safe, and will only produce new evaluators if no other invocation has
-   * already done so.
-   */
-  private <OutputT> TransformEvaluator<?> getTransformEvaluator(
+  private <OutputT> TransformEvaluator<?> createEvaluator(
       final AppliedPTransform<?, PCollection<OutputT>, ?> transform) {
-    // Key by the application and the context the evaluation is occurring in (which call to
-    // Pipeline#run).
-    Queue<BoundedReadEvaluator<OutputT>> evaluatorQueue =
-        (Queue<BoundedReadEvaluator<OutputT>>) sourceEvaluators.get(transform);
-    if (evaluatorQueue == null) {
-      evaluatorQueue = new ConcurrentLinkedQueue<>();
-      if (sourceEvaluators.putIfAbsent(transform, evaluatorQueue) == null) {
-        // If no queue existed in the evaluators, add an evaluator to initialize the evaluator
-        // factory for this transform
-        Bounded<OutputT> bound = (Bounded<OutputT>) transform.getTransform();
-        BoundedSource<OutputT> source = bound.getSource();
-        BoundedReadEvaluator<OutputT> evaluator =
-            new BoundedReadEvaluator<OutputT>(transform, evaluationContext, source);
-        evaluatorQueue.offer(evaluator);
-      } else {
-        // otherwise return the existing Queue that arrived before us
-        evaluatorQueue = (Queue<BoundedReadEvaluator<OutputT>>) sourceEvaluators.get(transform);
-      }
-    }
-    return evaluatorQueue.poll();
+    return new BoundedReadEvaluator<>(transform, evaluationContext);
   }
 
+  @Override
+  public void cleanup() {}
```

Comment:
This is the existing naming. If it's really important to you, I'm happy doing it in a separate PR, but it's not really relevant to this one.